pkg/pod/pod.go

```diff
@@ -322,6 +327,36 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		return nil, err
 	}
 
+	readonly := true
+	if config.FromContextOrDefaults(ctx).FeatureFlags.EnforceNonfalsifiability == config.EnforceNonfalsifiabilityWithSpire {
```

Comment:
nit: given that you do this check multiple times, might be useful to define a func in feature_flags.go and reuse it, for example:

https://github.com/tektoncd/pipeline/blob/a28eff704fd3f58ffe5179ce6e96afa031b8b781/pkg/apis/config/feature_flags.go#L326-L338